odoo/fields.py

```diff
@@ -2174,17 +2199,66 @@ def _setup_attrs(self, model, name):
             if 'selection_add' in field.args:
                 if self.related:
                     _logger.warning("%s: selection_add attribute will be ignored as the field is related", self)
+                self.extended = True
                 selection_add = field.args['selection_add']
                 assert isinstance(selection_add, list), \
                     "%s: selection_add=%r must be a list" % (self, selection_add)
                 assert values is not None, \
                     "%s: selection_add=%r on non-list selection %r" % (self, selection_add, self.selection)
+
+                new_options = [pair for pair in selection_add if pair not in self.selection]
```

Comment:
`pair` can be a singleton with an existing key, like `('foo',)`
```suggestion
                new_options = [pair for pair in selection_add if pair[0] not in values]
```
and it would be easier to get rid of the labels, for the code that comes below

Reply:
Yeah I realized this later on when a test crashed because I was unpacking two values and was only getting one, will do